### PR TITLE
Various read write fixes

### PR DESCRIPTION
**Pull request by HLWeil**
- Reading ISA from Contracts will now not return Assay and Study objects, when they are only registered in the investigation file, but have no corresponding folder/file in the filesystem.
- added a ARC.setFileSystem method, #250 
- investigation spreadsheet writer previously failed when it contained a study that was only registered, but no corresponding study object
- reworked test objects for full arc read-in

### CODE REVIEW

File: src/ISA/ISA.Spreadsheet/ArcInvestigation.fs [hunk 1]
```diff
@@ -194,7 +194,8 @@ module ArcInvestigation =
             yield  SparseRow.fromValues [contactsLabel]
             yield! Contacts.toRows (Some contactsLabelPrefix) (List.ofArray investigation.Contacts)
 
-            for study in investigation.RegisteredStudies do
+            for studyIdentifier in investigation.RegisteredStudyIdentifiers do
```
Comment (Freymaurer) [inline src/ISA/ISA.Spreadsheet/ArcInvestigation.fs:197]: What exactly is happening here? Can you add a few comments?

**Review by Freymaurer — COMMENTED** [review]
I want to add ARC level api for removing assay/study, which will return a real REMOVE contract and handle fs updating internal. Let me know what you think about it.

File: src/ARCtrl/ARC.fs [hunk 2]
```diff
@@ -71,6 +71,25 @@ type ARC(?isa : ISA.ArcInvestigation, ?cwl : CWL.CWL, ?fs : FileSystem.FileSyste
...
+    member this.RemoveAssay(assayIdentifier: string) =
+        if this.ISA.IsNone then failwith "Cannot remove assay from null ISA value."
+        this.ISA.Value.RemoveAssay(assayIdentifier)
+        let paths = this.FileSystem.Tree.ToFilePaths()
+        let assayFolderPath = Path.getAssayFolderPath(assayIdentifier)
+        let filteredPaths = paths |> Array.filter (fun p -> p.StartsWith(assayFolderPath) |> not)
+        this.SetFilePaths(filteredPaths)
+        Contract.createDelete(assayFolderPath)
+
+    member this.RemoveStudy(studyIdentifier: string) =
```
Comment (HLWeil) [inline src/ARCtrl/ARC.fs:85]: Also unregister?
Maybe we could do the naming like in the ArcCommander:

Delete -> Remove Object from filesystem by removing folder and file
Unregister -> Remove Registry reference from investigation file
Remove -> Delete + Unregister

Comment (HLWeil) [reply src/ARCtrl/ARC.fs:85]: Nvm, very good

File: src/ARCtrl/ARC.fs [hunk 3]
```diff
@@ -71,6 +71,25 @@ type ARC(?isa : ISA.ArcInvestigation, ?cwl : CWL.CWL, ?fs : FileSystem.FileSyste
 
     member this.FileSystem 
         with get() = _fs
+        and set(fs) = _fs <- fs
+
+    member this.RemoveAssay(assayIdentifier: string) =
```
Comment (HLWeil) [inline src/ARCtrl/ARC.fs:76]: These functions should contain update contracts for all registering objects (study,investigation)

Additional helpers:
- ContractCreator Type extensions for ArcInvestigation, ArcStudy and ArcAssay in Contract namespace

**Pull request merged**